jupyter_server/base/zmqhandlers.py

```diff
@@ -1,350 +1,9 @@
-            preferred_protocol = "v1.kernel.websocket.jupyter.org"
-        elif preferred_protocol == "":
-            preferred_protocol = None
-        selected_subprotocol = preferred_protocol if preferred_protocol in subprotocols else None
-        # None is the default, "legacy" protocol
-        return selected_subprotocol
-
-    def _on_zmq_reply(self, stream, msg_list):
-        # Sometimes this gets triggered when the on_close method is scheduled in the
-        # eventloop but hasn't been called.
-        if self.ws_connection is None or stream.closed():
-            self.log.warning("zmq message arrived on closed channel")
-            self.close()
-            return
-        channel = getattr(stream, "channel", None)
-        if self.selected_subprotocol == "v1.kernel.websocket.jupyter.org":
-            bin_msg = serialize_msg_to_ws_v1(msg_list, channel)
-            self.write_message(bin_msg, binary=True)
-        else:
-            try:
-                msg = self._reserialize_reply(msg_list, channel=channel)
-            except Exception:
-                self.log.critical("Malformed message: %r" % msg_list, exc_info=True)
-            else:
-                try:
-                    self.write_message(msg, binary=isinstance(msg, bytes))
-                except WebSocketClosedError as e:
-                    self.log.warning(str(e))
-
-
-class AuthenticatedZMQStreamHandler(ZMQStreamHandler, JupyterHandler):
-    def set_default_headers(self):
-        """Undo the set_default_headers in JupyterHandler
-
-        which doesn't make sense for websockets
-        """
-        pass
-
-    def pre_get(self):
-        """Run before finishing the GET request
-
-        Extend this method to add logic that should fire before
-        the websocket finishes completing.
-        """
-        # authenticate the request before opening the websocket
-        user = self.current_user
-        if user is None:
-            self.log.warning("Couldn't authenticate WebSocket connection")
-            raise web.HTTPError(403)
-
-        # authorize the user.
-        if not self.authorizer.is_authorized(self, user, "execute", "kernels"):
-            raise web.HTTPError(403)
-
-        if self.get_argument("session_id", None):
-            self.session.session = self.get_argument("session_id")
-        else:
-            self.log.warning("No session ID specified")
-
-    async def get(self, *args, **kwargs):
-        # pre_get can be a coroutine in subclasses
-        # assign and yield in two step to avoid tornado 3 issues
-        res = self.pre_get()
-        await res
-        res = super().get(*args, **kwargs)
-        await res
-
-    def initialize(self):
-        self.log.debug("Initializing websocket connection %s", self.request.path)
-        self.session = Session(config=self.config)
-
-    def get_compression_options(self):
-        return self.settings.get("websocket_compression_options", None)
+"""Add deprecation warning here.
```

Comment:
Still TODO?

Reply:
Ah, yes. I've now added a deprecation warning for this module.